server/src/middleware/csrf.js

```diff
@@ -0,0 +1,34 @@
+const assert = require("assert");
+const csrf = require("csurf");
+const mozlog = require("../logging").mozlog("csrf-middleware");
+const { captureRavenException } = require("../ravenclient");
+const { simpleResponse } = require("../responses");
+
+const config = require("../config").getProperties();
+
+const useSecureCsrfCookie = (config.expectProtocol && /^https$/.test(config.expectProtocol));
+
+exports.csrfProtection = csrf({cookie: {httpOnly: true, secure: useSecureCsrfCookie}});
+
+exports.csrf = function(req, res, next) {
+  // The cookies library doesn't detect duplicates; check manually
+  let rawCookies = req.get("cookie") || "";
+  let pairs = rawCookies.split(";");
+  let csrfTokens = pairs.filter(item => item.match(/_csrf=/));
+  if (csrfTokens.length > 1) {
+    let exc = new Error("Duplicate CSRF cookies");
+    exc.headerValue = rawCookies;
+    captureRavenException(exc);
+    return simpleResponse(res, "Bad request", 400);
+  }
+  req.cookies._csrf = req.cookies.get("_csrf"); // csurf expects a property
```

Comment:
I thought we wanted to change the cookie name from '_csrf' to '__Host-csrf'?

Reply:
We do.  I think changing prefixes in a separate PR will be less likely to break things and didn't realize the `__Host-` requires the secure attribute (so we'd need to something like change prefixes depending on whether we're setting secure or not).

https://tools.ietf.org/html/draft-ietf-httpbis-cookie-prefixes-00#section-3.2

Reply:
I'm guessing local development is the only reason to allow non-https server URLs. Maybe this issue is a good excuse to force us to get https working locally

Reply:
For a while we did require https locally, it was pretty complicated to setup. I felt good when I was able to remove it.

Reply:
Yeah, I just noticed #2265. Webpack has a pretty nice https out of the box experience, but it can be pretty painful to setup manually, accept the certs in the temporary profile, etc.

Reply:
Followup in #3455